l10n_ch_sepa_credit_transfer/wizard/export_sepa.py

```diff
@@ -0,0 +1,231 @@
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.osv import orm, fields
+from openerp.tools.translate import _
+from lxml import etree
+
+
+class BankingExportSepaWizard(orm.TransientModel):
+    _inherit = 'banking.export.sepa.wizard'
+
+    def create_sepa(self, cr, uid, ids, context=None):
+        '''
+        Creates the SEPA Credit Transfer file. That's the important code !
+        '''
+        if context is None:
+            context = {}
+        sepa_export = self.browse(cr, uid, ids[0], context=context)
+        pain_flavor = sepa_export.payment_order_ids[0].mode.type.code
+        convert_to_ascii = \
+            sepa_export.payment_order_ids[0].mode.convert_to_ascii
+        if pain_flavor == 'pain.001.001.02':
+            bic_xml_tag = 'BIC'
+            name_maxsize = 70
+            root_xml_tag = 'pain.001.001.02'
+        elif pain_flavor == 'pain.001.001.03':
+            bic_xml_tag = 'BIC'
+            # size 70 -> 140 for <Nm> with pain.001.001.03
+            # BUT the European Payment Council, in the document
+            # "SEPA Credit Transfer Scheme Customer-to-bank
+            # Implementation guidelines" v6.0 available on
+            # http://www.europeanpaymentscouncil.eu/knowledge_bank.cfm
+            # says that 'Nm' should be limited to 70
+            # so we follow the "European Payment Council"
+            # and we put 70 and not 140
+            name_maxsize = 70
+            root_xml_tag = 'CstmrCdtTrfInitn'
+        elif pain_flavor == 'pain.001.001.04':
+            bic_xml_tag = 'BICFI'
+            name_maxsize = 140
+            root_xml_tag = 'CstmrCdtTrfInitn'
+        elif pain_flavor == 'pain.001.001.05':
+            bic_xml_tag = 'BICFI'
+            name_maxsize = 140
+            root_xml_tag = 'CstmrCdtTrfInitn'
+        # added pain.001.003.03 for German Banks
+        # it is not in the offical ISO 20022 documentations, but nearly all
+        # german banks are working with this instead 001.001.03
+        elif pain_flavor == 'pain.001.003.03':
+            bic_xml_tag = 'BIC'
+            name_maxsize = 70
+            root_xml_tag = 'CstmrCdtTrfInitn'
+        elif pain_flavor == 'pain.001.003.03.ch.02':
```

Comment:
I guess there is a typo here:

xml data file is defining: `pain.001.001.03.ch.02`